File: https:/github.com/rajesh-kayal-dev/Leetcode.git/153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.cpp
// Last updated: 6/15/2026, 10:01:56 AM
class Solution {
public:
    int findMin(vector<int>& arr) {
        int start=0,end=arr.size()-1,ans=arr[0];
        while(start<=end)
        {
            int mid= start+ (end - start)/2;
            if(arr[mid]>=arr[0]){
                start=mid+1;
            }else{
                ans=arr[mid];
                end=mid-1;
            }
        }
        return ans;
    }
};

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.java
// Last updated: 6/15/2026, 10:01:48 AM
class Solution {
    public int findMin(int[] nums) {
        int low = 0;
        int high = nums.length -1;

        while(low < high){
            int mid = low + (high - low) /2;

            //if mid element > rightmost element then min at the right half
            if(nums[mid] > nums[high]){
                low = mid + 1;
            }else{
                high = mid;
            }
        }
        return nums[low]; //or nums[high] or at the min point
    }
}